applications/SwimmingDEMApplication/custom_elements/qsvms_dem_coupled.cpp

```diff
@@ -0,0 +1,859 @@
+        rMassMatrix = ZeroMatrix(LocalSize, LocalSize);
+        TElementData data;
+        data.Initialize(*this, rCurrentProcessInfo);
+        // Get the element's geometric parameters
+        double Area = data.Weight;
+
+        // Calculate this element's fluid properties
+
+        // Add 'classical' mass matrix (lumped)
+        double Coeff = Area / NumNodes; //Optimize!
+
+        this->CalculateLaplacianLumpedMassMatrix(rMassMatrix, Coeff);
+    }
+
+    /// Write the divergence of the advective velocity evaluated at this point to an array
+    /**
+     * Writes the value of the divergence of the advective velocity evaluated at a point inside
+     * the element to a double
+     * @param rAdvVelDiv: Output array
+     * @param rShapeDeriv: Derivatives of shape functions evaluated at the integration point
+     * @param Step: The time Step
+     */
+template<class TElementData>
+void QSVMSDEMCoupled<TElementData>::GetAdvectiveVelDivergence(double & rAdvVelDiv,
+                                                            const BoundedMatrix<double, NumNodes, Dim >& rDN_DX)
+    {
+        rAdvVelDiv = 0.0;
+
+        for (unsigned int i = 1; i < NumNodes; ++i){ // loop over nodes
+            const array_1d< double, 3 > vel_at_nodes = this->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY) - this->GetGeometry()[i].FastGetSolutionStepValue(MESH_VELOCITY);
+
+            for (unsigned int d = 1; d < Dim; ++d){
+                // loop over components
+                rAdvVelDiv += vel_at_nodes[d] * rDN_DX(i, d);
+            }
+
+        }
+
+    }
+
+template<class TElementData>
+void QSVMSDEMCoupled<TElementData>::CalculateLaplacianLumpedMassMatrix(MatrixType& rLHSMatrix,
+                                                                        const double Mass)
+    {
+        unsigned int DofIndex = 0;
+        for (unsigned int iNode = 0; iNode < NumNodes; ++iNode)
+        {
+            for (unsigned int d = 0; d < Dim; ++d)
+            {
+                rLHSMatrix(DofIndex, DofIndex) += Mass;
+                ++DofIndex;
+            }
+        }
+    }
+
+template<class TElementData>
+void QSVMSDEMCoupled<TElementData>::AddRHSLaplacian(VectorType& F,
+                                        const BoundedMatrix<double, NumNodes, Dim>& rDN_DX,
+                                        const double Weight)
+    {
+        double Coef = Weight;
+        array_1d<double, 3 > Velocity(3, 0.0);
+
+        int LocalIndex = 0;
+        int LocalNodalIndex = 0;
+
+        for (unsigned int i = 0; i < NumNodes; ++i)
+        {
+            Velocity = this->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY);
+            for (unsigned int d = 0; d < Dim; ++d)
+            {
+                F[LocalIndex++] -= Coef * rDN_DX(LocalNodalIndex, d) * Velocity[d] * rDN_DX(i, d);
+            }
+            LocalNodalIndex++;
+        }
+    }
+
+template<class TElementData>
+void QSVMSDEMCoupled<TElementData>::AddMomentumRHS(VectorType& F,
```

Comment:
```suggestion
void QSVMSDEMCoupled<TElementData>::AddMomentumRHS(VectorType& rDescriptiveVectorName,
```
Plase use descriptive argument names.